File: CSharp/MongoDB/CRUD.Client/MongoDBClientBuilder.cs
using MongoDB.Driver;

namespace MongoDB.CRUD;

public class MongoDBClientBuilder : IMongoDBClientBuilder
{
    public MongoDBClientBuilder(MongoClientSettings mongoClientSettings)
    {
        MongoClientSettings = mongoClientSettings;
    }

    private MongoClientSettings MongoClientSettings { get; }

    public IMongoClient Build()
    {
        return new MongoClient(MongoClientSettings);
    }
}

File: Examples/PHP/Containers/Laravel-MongoDB-REST/app/Services/MongoDB/CRUD/MongoDB_Client_Builder.php
<?php

namespace App\Services\MongoDB\CRUD;

class MongoDB_Client_Builder implements IMongoDB_Client_Builder
{
    private $connectionString;

    public function __construct(MongoDB_Config $config)
    {
        $this->connectionString = $config->toUri();
    }

    public function build(): \MongoDB\Client
    {
        return new \MongoDB\Client($this->connectionString);
    }
}